File: Java/HLC04_EjemplosCodigo1_5/03_Ud4_ServicioDescargaAsyncTask/app/src/main/java/com/pms/a03_ud4_serviciodescargaasynctask/MainActivity.java
package com.pms.a03_ud4_serviciodescargaasynctask;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public void arrancarServicio(View view) {
        startService(new Intent(getBaseContext(), ServiceDescarga.class));
    }

    public void pararServicio(View view) {
        stopService(new Intent(getBaseContext(), ServiceDescarga.class));
    }
}

File: Kotlin/HLC04_EjemplosCodigo1_5/03b_Ud4_ServicioDescargaAsyncTask/app/src/main/java/com/pms/a03b_ud4_serviciodescargaasynctask/MainActivity.kt
package com.pms.a03b_ud4_serviciodescargaasynctask

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun arrancarServicio(view: View?) {
        startService(Intent(baseContext, ServiceDescarga::class.java))
    }

    fun pararServicio(view: View?) {
        stopService(Intent(baseContext, ServiceDescarga::class.java))
    }
}